src/include/ipc/probe.h

```diff
@@ -0,0 +1,183 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+/**
+ * \brief Definitions of shifts and masks for format encoding in probe
+ *	  extraction stream
+ *
+ * Audio format from extraction probes is encoded as 32 bit value. Following
+ * graphic explains encoding.
+ *
+ * A|BBBB|CCCC|DDDD|EEEEE|FF|GG|H|I|J|XXXXXXX
+ * A - 1 bit - Specifies Type Encoding - 1 for Standard encoding
+ * B - 4 bits - Specify Standard Type - 0 for Audio, 1 for Video
+ * C - 4 bits - Specify Audio format - 0 for PCM, 1 for MP3
+ * D - 4 bits - Specify Sample Rate - value enumerating standard sample rates:
+ *				      8000 Hz		= 0x0
+ *				      11025 Hz		= 0x1
+ *				      12000 Hz		= 0x2
+ *				      16000 Hz		= 0x3
+ *				      22050 Hz		= 0x4
+ *				      24000 Hz		= 0x5
+ *				      32000 Hz		= 0x6
+ *				      44100 Hz		= 0x7
+ *				      48000 Hz		= 0x8
+ *				      64000 Hz		= 0x9
+ *				      88200 Hz		= 0xA
+ *				      96000 Hz		= 0xB
+ *				      128000 Hz		= 0xC
+ *				      176400 Hz		= 0xD
+ *				      192000 Hz		= 0xE
+ *				      none of the above = 0xF
+ * E - 5 bits - Specify Number of Channels minus 1
+ * F - 2 bits - Specify Sample Size, number of valid sample bytes minus 1
+ * G - 2 bits - Specify Container Size, number of container bytes minus 1
```

Comment:
how do you represent 24 bits in a 32-bit container? This could be left or right aligned?

Reply:
It's exactly as the format played within the audio pipeline, so right-aligned if I'm not mistaken.
We don't do any corrections, or shifting, re-encoding of data whatsoever.

Reply:
I am pretty sure Windows assumes (or used to assume) a left-aligned representation. That was one of the changes for HDMI some 6+ years ago.

Reply:
While true, it isn't really important here, as we're straight-up copying data from audio pipeline, without any changes to the data, including container size and alignment within it. As such, on Windows audio data in the pipeline would be aligned properly for Windows machine, and likewise on Linux. I guess that my previous statement about alignment wasn't dead-on, as I didn't consider Windows in my answer, as we're not supporting it yet.

Reply:
obsolete documentation removed